### PR TITLE
Continuous transforms

### PR DESCRIPTION
**Pull request by usmanm**

### CODE REVIEW

**Comment by derekjn** [comment]
Am I doing something dumb here?

```
create continuous view v as select x::integer from even_stream;
create continuous transform tv as
    select x::integer from stream where x % 2 = 0
    then execute procedure pipeline_stream_insert('even_stream');

insert into stream (x) values (2);
select * from v;
 x 
---
      <-- null
(1 row)
```

**Comment by usmanm** [comment]
synchronous_stream_insert doesn't work right now. Could that be an issue?

**Comment by derekjn** [comment]
I don't think so. If I insert another row, there are two null rows in the CV, so it seems to be reading something.

**Comment by usmanm** [comment]
I'll look into it in a little bit when I get home.

On Saturday, February 27, 2016, Derek Nelson <notifications@github.com>
wrote:

> I don't think so. If I insert another row, there are two null rows in the
> CV, so it seems to be reading something.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pipelinedb/pipelinedb/pull/1393#issuecomment-189769037>
> .
>


-- 
Usman

**Comment by usmanm** [comment]
Should be fixed now.

**Comment by derekjn** [comment]
This is awesome man. The code looks good as far as I can tell, I'm going to continue spending some time functionally testing it though. I'll also add a few tests to this branch.

**Comment by derekjn** [comment]
`catversion.h:CATALOG_VERSION_NO` needs to be bumped.

**Comment by usmanm** [comment]
Also need to add support for `pg_dump` and `pg_restore`.

**Comment by derekjn** [comment]
I'm also getting `NULL` rows in continuous view reading from a continuous xform doing an STJ:

```
create continuous transform x0 as
    select t.s as s from raw_stream s join t on t.x = s.x::integer
    then execute procedure pipeline_stream_insert('stream');
create continuous view v0 as select s::text from stream;
create continuous view v1 as select s::text from stream;

insert into stream (x) values (0);
select * from v0;
 s 
---
 
(1 row)
```

File: src/backend/pipeline/transformReceiver.c [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+	MemoryContextSwitchTo(old);
+}
+
+static void
+transform_destroy(DestReceiver *self)
+{
+	TransformState *t = (TransformState *) self;
+	pfree(t);
+
+	elog(LOG, "destroy");
```
Comment (derekjn) [inline src/backend/pipeline/transformReceiver.c]: Remove

**Comment by derekjn** [comment]
re: my previous comment, I forgot to mention that before doing the stream insertion I did:

```
create table t (x integer, s text);
insert into t (x, s) values (0, 'zero');
insert into t (x, s) values (1, 'one');
```

Also, it appears that `transform_receive` isn't even being called after the stream insertion. It doesn't appear that `init_query_state` is being called for the xform either.

**Comment by derekjn** [comment]
Disregard previous comments about null rows, I was doing something dumb.

**Pull request closed without merging**